src/js/script.js

```diff
@@ -185,11 +187,92 @@ const select = {
         }
       }
     }
+
+    // multiply price by amount
+    price *= thisProduct.amountWidget.value;
+
     // update calculated price in the HTML
     thisProduct.priceElem.innerHTML = price;
     }
+
+    initAmountWidget(){
+      const thisProduct = this;
+
+      thisProduct.amountWidget = new AmountWidget(thisProduct.amountWidgetElem);
+      thisProduct.amountWidgetElem.addEventListener('updated',function(){
+        thisProduct.processOrder();
+      });
+    }
   }
   
+  class AmountWidget{
+    construtor(element){
+      const thisWidget = this;
+      console.log('AmountWidget', thisWidget);
+      console.log('constructor arguments:', element);
+
+      thisWidget.getElements(element);
+      thisWidget.setValue(thisWidget.input.value);
+      thisWidget.initActions();
+    }
+
+    getElements(element){
+      const thisWidget = this;
+    
+      thisWidget.element = element;
+      thisWidget.input = thisWidget.element.querySelector(select.widgets.amount.input);
+      thisWidget.linkDecrease = thisWidget.element.querySelector(select.widgets.amount.linkDecrease);
+      thisWidget.linkIncrease = thisWidget.element.querySelector(select.widgets.amount.linkIncrease);
+    }
+
+    setValue(value){
+      const thisWidget = this;
+      const newValue = parseInt(value);
+      /* TODO: Add validation */
+
+      if (thisWidget.value !== newValue) {
+        thisWidget.value = newValue;
+      }
+      if (thisWidget.value !== newValue && !isNaN(newValue)) {
+        thisWidget.value = newValue;
+      }
+      if (thisWidget.value >= settings.amountWidget.defaultMin){
+        thisWidget.value = newValue;
+      }
+      if (thisWidget.value <= settings.amountWidget.defaultMax){
+        thisWidget.value = newValue;
+      }
+
+      thisWidget.value = newValue;
+      thisWidget.input.value = thisWidget.value;
+
+      thisWidget.announce();
+    }
+
+    initActions (){
+      const thisWidget = this;
+      thisWidget.input.addEventListener('change', function(){
+        thisWidget.value = thisWidget.input.value;
+      });
+
+      thisWidget.linkDecrease.addEventListener('click',function(event){
+        event.preventDefault();
+        thisWidget.setValue(thisWidget.value - 1)
+      });
+
+      thisWidget.linkDecrease.addEventListener('click',function(event){
```

Comment:
A tutaj zapewne chodziło o `linkIncrease` skoro dodajesz 1 do wartości ;)
```suggestion
      thisWidget.linkIncrease.addEventListener('click',function(event){
```